src/screens/HyperSwitch/Connectors/ConnectPayPalFlow/ConnectPayPal.res

```diff
@@ -0,0 +1,534 @@
+              {errorPageDetails.headerText->React.string}
+            </p>
+          </UIUtils.RenderIf>
+          <UIUtils.RenderIf condition={errorPageDetails.subText->Js.String2.length > 0}>
+            <p className=p1RegularTextClass> {errorPageDetails.subText->React.string} </p>
+          </UIUtils.RenderIf>
+        </div>
+        <UIUtils.RenderIf condition={errorPageDetails.buttonText->Belt.Option.isSome}>
+          <PayPalCreateNewAccountModal
+            butttonDisplayText={errorPageDetails.buttonText->Belt.Option.getWithDefault("")}
+            actionUrl
+            setScreenState
+          />
+        </UIUtils.RenderIf>
+      </div>
+      <UIUtils.RenderIf condition={errorPageDetails.refreshStatusText->Belt.Option.isSome}>
+        <div className="flex gap-2">
+          <p className=p1RegularTextClass>
+            {errorPageDetails.refreshStatusText->Belt.Option.getWithDefault("")->React.string}
+          </p>
+          <p
+            className={`${p1RegularTextClass} !text-blue-700 !opacity-80 cursor-pointer`}
+            onClick={_ => getStatus()->ignore}>
+            {"Refresh Status"->React.string}
+          </p>
+        </div>
+      </UIUtils.RenderIf>
+    </div>
+  }
+}
+
+@react.component
+let make = (
+  ~connector,
+  ~connectorAccountFields,
+  ~selectedConnector,
+  ~connectorMetaDataFields,
+  ~connectorWebHookDetails,
+  ~isUpdateFlow,
+  ~setInitialValues,
+  ~handleConnectorConnected,
+  ~initialValues,
+  ~setShowModal,
+  ~showVerifyModal,
+  ~setShowVerifyModal,
+  ~verifyErrorMessage,
+  ~setVerifyDone,
+  ~handleStateToNextPage,
+  ~connectorLabelDetailField,
+  ~showModal,
+) => {
+  open APIUtils
+  open LogicUtils
+  let url = RescriptReactRouter.useUrl()
+  let showToast = ToastState.useShowToast()
+
+  let connectorValue = isUpdateFlow
+    ? url.path->Belt.List.toArray->Belt.Array.get(1)->Belt.Option.getWithDefault("")
+    : url.search
+      ->getDictFromUrlSearchParams
+      ->Dict.get("connectorId")
+      ->Belt.Option.getWithDefault("")
+
+  let (connectorId, setConnectorId) = React.useState(_ => connectorValue)
+  let updateDetails = useUpdateMethod(~showErrorToast=false, ())
+  let isRedirectedFromPaypalModal =
+    url.search
+    ->getDictFromUrlSearchParams
+    ->Dict.get("is_back")
+    ->Belt.Option.getWithDefault("")
+    ->getBoolFromString(false)
+
+  let (screenState, setScreenState) = React.useState(_ => PageLoaderWrapper.Success)
+  let (configuartionType, setConfigurationType) = React.useState(_ => PayPalFlowTypes.NotSelected)
+  let (actionUrl, setActionUrl) = React.useState(_ => "")
+
+  let (setupAccountStatus, setSetupAccountStatus) = Recoil.useRecoilState(
+    HyperswitchAtom.paypalAccountStatusAtom,
+  )
+  let (suggestedAction, suggestedActionExists) = ConnectorUtils.getSuggestedAction(
+    ~verifyErrorMessage,
+    ~connector,
+  )
+
+  let onSubmitMain = async values => {
+    open ConnectorUtils
+    open PayPalFlowUtils
```

Comment:
It;s open and called multiple places just check once

Reply:
Resolved